### PR TITLE
LARS float support cleanup patch (for PR #3133)

### PR DESCRIPTION
**Pull request by c4goldsw**
Cleanup for #3313

### CODE REVIEW

File: src/shogun/labels/DenseLabels.h [hunk 1]
```diff
@@ -123,7 +123,7 @@ class CDenseLabels : public CLabels
 		ST get_label_t(int32_t idx)
 		{
 			int32_t real_num=m_subset_stack->subset_idx_conversion(idx);
-			ASSERT(m_labels.vector && idx<get_num_labels())
+			REQUIRE(m_labels.vector && idx<get_num_labels(), "Attempt to get label: index out of bounds")
```
Comment (karlnapf) [inline src/shogun/labels/DenseLabels.h]: Please write proper [error messages](https://github.com/shogun-toolbox/shogun/wiki/Assertions)

File: src/shogun/labels/DenseLabels.h [hunk 2]
```diff
@@ -156,7 +156,7 @@ class CDenseLabels : public CLabels
 				return get_labels_copy_t<ST>();
 
 			SGVector<ST> labels_copy(m_labels.vlen);
-			for(int i = 0; i < m_labels.vlen; ++i)
+			for(index_t i = 0; i < m_labels.vlen; ++i)
```
Comment (karlnapf) [inline src/shogun/labels/DenseLabels.h]: @lisitsyn how does your range thing work?

File: src/shogun/regression/LeastAngleRegression.cpp [hunk 3]
```diff
@@ -39,13 +39,15 @@ CLeastAngleRegression::~CLeastAngleRegression()
 	
 }
 
+#ifndef SWIG
```
Comment (karlnapf) [inline src/shogun/regression/LeastAngleRegression.cpp]: you only need to do that in the header

File: src/shogun/regression/LeastAngleRegression.cpp [hunk 4]
```diff
@@ -123,12 +120,13 @@ bool CLeastAngleRegression::train_machine(CFeatures* data)
 	else
 		SG_ERROR("Feature-type (%d) must be of type F_SHORTREAL (%d), F_DREAL (%d) or F_LONGREAL (%d).\n", 
 			data->get_feature_type(), F_SHORTREAL, F_DREAL, F_LONGREAL)
+
+	return false;
```
Comment (karlnapf) [inline src/shogun/regression/LeastAngleRegression.cpp]: false?

File: src/shogun/regression/LeastAngleRegression.h [hunk 5]
```diff
@@ -124,10 +124,9 @@ class CLeastAngleRegression: public CLinearMachine
 	
 	void switch_w(int32_t num_variable)
 	{
-		if (w.vlen <= 0)
-			SG_ERROR("Please train the model before updating its parameters")
-		if (size_t(num_variable) >= m_beta_idx.size() || num_variable < 0)
-			SG_ERROR("cannot switch to an estimator of %d non-zero coefficients", num_variable)
+		REQUIRE(w.vlen > 0,"Please train the model before updating its parameters")
```
Comment (karlnapf) [inline src/shogun/regression/LeastAngleRegression.h]: newline missing.
Also, is this a message for the user? Then it needs to be more concrete, i.e. how to solve it?

File: src/shogun/regression/LeastAngleRegression.h [hunk 6]
```diff
@@ -124,10 +124,9 @@ class CLeastAngleRegression: public CLinearMachine
...
+		REQUIRE(size_t(num_variable) < m_beta_idx.size() && num_variable >= 0,
+			"cannot switch to an estimator of %d non-zero coefficients", num_variable)
```
Comment (karlnapf) [inline src/shogun/regression/LeastAngleRegression.h]: Cannot

Comment (karlnapf) [inline src/shogun/regression/LeastAngleRegression.h]: newline missing

File: src/shogun/regression/LeastAngleRegression.h [hunk 7]
```diff
@@ -210,14 +208,18 @@ class CLeastAngleRegression: public CLinearMachine
 	static void plane_rot(ST x0, ST x1,
 		ST &y0, ST &y1, SGMatrix<ST> &G);
 
+	#ifndef SWIG
```
Comment (karlnapf) [inline src/shogun/regression/LeastAngleRegression.h]: yes, but remove the newline in between

File: tests/unit/regression/lars_unittest.cc [hunk 8]
```diff
@@ -252,8 +252,23 @@ void lars_n_less_than_d_feature_test_templated()
...
-	lars->train(features);
+
+	//Catch exceptions thrown when training, clean up
+	try
+	{
+		lars->train(features);
+	}
+	catch(...)
+	{
+		SG_UNREF(lars);
```
Comment (karlnapf) [inline tests/unit/regression/lars_unittest.cc]: Did you check this for potential segfaults?

**Comment by karlnapf** [comment]
Can you squash the commits to a single one?

**Comment by karlnapf** [comment]
I will be away for quite some time now, so leaving @vigsterkr to merge this

File: src/shogun/regression/LeastAngleRegression.cpp (hunk 4 above)
Comment (c4goldsw) [inline src/shogun/regression/LeastAngleRegression.cpp]: I'm just including a return statement so that there are no compiler warnings.

**Comment by c4goldsw** [comment]
@karlnapf Ok :)

File: tests/unit/regression/lars_unittest.cc (hunk 8 above)
Comment (c4goldsw) [inline tests/unit/regression/lars_unittest.cc]: Yes, using cout

File: src/shogun/regression/LeastAngleRegression.cpp (hunk 4 above)
Comment (vigsterkr) [inline src/shogun/regression/LeastAngleRegression.cpp]: `false` is good... better to be safe than sorry :P and anyhow it's almost as if you put it into `else` branch...

File: tests/unit/regression/lars_unittest.cc (hunk 8 above)
Comment (vigsterkr) [inline tests/unit/regression/lars_unittest.cc]: it's quite painful to say this but this is a good place for goto :P
but then again you wanna throw the exception :)

**Comment by vigsterkr** [comment]
@c4goldsw thnx heaps for the patch! lets merge it and see what the memchecker does

**Pull request merged**